dev-guide/api-conventions.md

```diff
@@ -109,6 +109,24 @@ This means, that any end user fetching the resource from the API can observe tha
 This has the effect of helping our users to understand how they might be able to configure their cluster, without
 having to search for features or review API schemas within the product docs.
 
+### Only one phrasing for each idea
+
+Each idea should have a single possible expression in the API structures, without having multiple ways to say the same thing.
```

Comment:
The example in the commit message would help the reader understand how the principle might be applied in the context of Kubernetes API modelling

Reply:
I've worked that example in with b62f09e0 -> a3f21140, but if anyone has a real-world example that doesn't violate another of our conventions, that would be useful.

Reply:
I agree that a real-world example would be useful.  As it is, this new guideline seemingly proscribes only choices that are already proscribed by other guidelines.  In addition, it proscribes choices that other guidelines recommend, such as discriminated unions with empty union members.

Reply:
> In addition, it proscribes choices that other guidelines recommend, such as discriminated unions with empty union members.

Good point.  I'm unclear how [the current discriminated union pointer requirement][1] squares with [the no-pointer guideline][2].  But given that that conflict already exists today, I don't see how my proposed change is making it worse.

[1]: https://github.com/openshift/enhancements/blob/890debb43169f0458246c3e19976a4c4bdda5b30/dev-guide/api-conventions.md#writing-a-union-in-go
[2]: https://github.com/openshift/enhancements/blob/890debb43169f0458246c3e19976a4c4bdda5b30/dev-guide/api-conventions.md#pointers

Reply:
The discriminated union pointer is an exception that's required, we should make a note that this is an exception to the rule I guess